opsdroid/connector/slack/__init__.py

```diff
@@ -62,6 +62,9 @@ def __init__(self, config):
             _LOGGER.error(error)
             _LOGGER.error("Failed to connect to Slack, retrying in 10")
             await self.reconnect(10)
+        except Exception:
```

Comment:
Perhaps instead of catching a broad `Exception` and reraising it you could use a `finally` block.

Reply:
I don't think `finally` will work here, because `finally` will get called even if no exception is raised, which we don't want.  I don't think the broad `Exception` here is necessarily an anti-pattern, since we do want do to some cleanup if something goes wrong, and we re-raise the Exception after doing that cleanup.

It does seem like #717 is semi-related here, and it seems like having a `disconnect` method here is the "right" approach for handling the cleanup, so I slightly modified the PR; but I don't have a better idea for how to actually handle arbitrary exceptions.  Happy to hear suggestions, though :)